lib/src/config/config.rs

```diff
@@ -494,6 +626,25 @@ impl ClientConfigBuilder {
         self
     }
 
+    /// Adds a custom seed node or seed list
+    pub fn seed<S: Into<Seed>>(&mut self, seed: S) -> &mut Self {
+        let mut seeds = self.seeds.get_or_insert_with(Default::default);
+        seeds.push(seed.into());
+        self
+    }
+
+    /// Adds a custom seed node
+    pub fn seed_uri<U: Into<PeerUri>>(&mut self, uri: U) -> &mut Self {
+        self.seed(Seed::new_peer(uri.into()))
+    }
+
+    /// Adds a custom seed url
+    pub fn seed_list(&mut self, url: Url, public_key_opt: Option<PublicKey>) -> &mut Self {
+        self.seed(Seed::new_list(SeedList::new(url, public_key_opt)))
+    }
+
+    /// Sets the validator config. Since there is no configuration for validators (except key file)
+    /// yet, this will just enable the validator.
     pub fn validator(&mut self) -> &mut Self {
```

Comment:
This probably needs to go behind a feature flag as the `validator` field itself is behind a feature flag, too.

The `Some(Some(ValidatorConfig{}))` part looks strange, but I guess the extra Option comes from the builder.
`ValidatorConfig{}` can be simplified to `ValidatorConfig`.